src/component/CodeBlock.tsx

```diff
@@ -13,18 +17,79 @@ darkTheme["hljs-selection"] = {
   backgroundColor: "#3a404b" // https://github.com/atom/one-dark-ui/blob/master/styles/ui-variables.less#L32
 };
 
+const useStyles = makeStyles(theme => ({
+  copyButton: {
+    float: 'left'
+  },
+  hidden: {
+    display: 'none'
+  },
+  copied: {
+    opacity: 0,
+    transition: 'opacity 0.3s'
+  }
+}));
+
+
 function CodeBlock(props: SyntaxHighlighterProps) {
   const darkMode = useDarkMode();
+  const classes = useStyles(darkMode);
+
+  function onContainerEnter() {
+    let newState = Object.assign({}, state)
+    newState.showCopy = true
+    setState( newState );
+  }
+
+  function onContainerLeave() {
+    let newState = Object.assign({}, state)
+    newState.showCopy = false
+    setState( newState );
+  }
+
+  function onCopy() {
+    let newState = Object.assign({}, state)
+    newState.copied = true
+    setState( newState );
+    setTimeout(() => {
+      newState = Object.assign({}, state)
+      newState.copied = false;
+      setState(newState)
+    }, 300)
+  }
+
+  const [state, setState] = React.useState({
+    showCopy: false,
+    copied: false,
+  });
+
   return (
-    <SyntaxHighlighter
-      style={darkMode ? darkTheme : lightTheme}
-      language={props.language || "js"}
-      showLineNumbers={props.showLineNumbers || false}
-      wrapLines={true}
-      lineProps={props.lineProps}
-    >
-      {props.value}
-    </SyntaxHighlighter>
+    <Container
```

Comment:
This container adds an extra margin on the left and right - I don't think we want that.